flink-streaming-connectors/flink-connector-cassandra/src/test/java/org/apache/flink/streaming/connectors/cassandra/CassandraConnectorITCase.java

```diff
@@ -173,39 +157,49 @@ public static void startCassandra() throws IOException {
 			cassandra.start();
 		}
 
-		try {
-			Thread.sleep(1000 * 10);
-		} catch (InterruptedException e) { //give cassandra a few seconds to start up
+		// give cassandra a few seconds to start up
+		long start = System.currentTimeMillis();
+		long deadline = start + 1000 * 10;
+		while (System.currentTimeMillis() < deadline) {
+			try {
+				Thread.sleep(1000);
+			} catch (InterruptedException ignored) {
+			}
 		}
 
-		cluster = builder.getCluster();
-		session = cluster.connect();
+		// start establishing a connection within 30 seconds
+		start = System.currentTimeMillis();
```

Comment:
I would suggest to use `System.nanoTime()`, because `System.curremtTimeMillis` is not stable. Seems especially unstable on Travis.